### PR TITLE
Core: Remove `storyStoreV7` feature flag

### PR DESCRIPTION
**Pull request by ndelangen**
Relates to https://github.com/storybookjs/storybook/issues/25117

## What I did

Step 2 of https://github.com/storybookjs/storybook/pull/24655

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by ndelangen** [comment]
@SocketSecurity ignore @vitejs/plugin-vue@4.4.0
@SocketSecurity ignore @babel/preset-modules@0.1.6
@SocketSecurity ignore piscina@4.0.0
@SocketSecurity ignore loupe@2.3.7

File: docs/api/main-config-features.md [hunk 1]
```diff
@@ -107,26 +67,4 @@ Filter args with a "target" on the type from the render function.
...
-
-## On-demand story loading
-
-As your Storybook grows, it gets challenging to load all of your stories performantly, slowing down the loading times and yielding a large bundle. Out of the box, Storybook loads your stories on demand rather than during boot-up to improve the performance of your Storybook. If you need to load all of your stories during boot-up, you can disable this feature by setting the `storyStoreV7` feature flag to `false` in your configuration as follows:
-
-<!-- prettier-ignore-start -->
-
-<CodeSnippets
-  paths={[
-    'common/main-config-features-story-store-v7.js.mdx',
```
Comment (kylegach) [inline docs/api/main-config-features.md:120]: Doing a Find in project for this snippet filename also yields these areas to also be removed:

- https://github.com/storybookjs/storybook/blob/next/docs/configure/index.md?plain=1#L57-L58
- https://github.com/storybookjs/storybook/blob/next/docs/configure/index.md?plain=1#L134-L154
- https://github.com/storybookjs/storybook/blob/next/docs/configure/webpack.md?plain=1#L54-L68

There's also this:

- https://github.com/storybookjs/storybook/blob/next/docs/migration-guide.md?plain=1#L123-L136

But the better move there would be updating the entire guide to be `8.0`-focused, which should probably happen in another PR.

-----

`storyStoreV7` and `buildStoriesJson` also have multiple additional instances throughout the docs that should be removed.

**Comment by yannbf** [comment]
Please add migration notes that cover all breaking changes, especially when it comes to APIs which could break addons 🙏 

These two issues might give a bit of light, but there is definitely much more to cover:
https://github.com/storybookjs/storybook/issues/25367
https://github.com/storybookjs/storybook/issues/25365

File: code/lib/preview-api/src/modules/preview-web/Preview.tsx [hunk 2]
```diff
@@ -185,15 +163,11 @@ export class Preview<TRenderer extends Renderer> {
   }
 
   // If initialization gets as far as the story index, this function runs.
-  initializeWithStoryIndex(storyIndex: StoryIndex): PromiseLike<void> {
+  initializeWithStoryIndex(storyIndex: StoryIndex): void {
```
Comment (shilman) [inline code/lib/preview-api/src/modules/preview-web/Preview.tsx:166]: Shouldn't this be async?

File: code/lib/preview-api/src/modules/preview-web/PreviewWeb.integration.test.ts [hunk 3]
```diff
@@ -76,7 +73,7 @@ beforeEach(() => {
   vi.mocked(WebView.prototype).prepareForStory.mockReturnValue('story-element' as any);
 });
 
-describe('PreviewWeb', () => {
+describe.skip('PreviewWeb', () => {
```
Comment (shilman) [inline code/lib/preview-api/src/modules/preview-web/PreviewWeb.integration.test.ts:76]: What's the plan here? @tmeasday to address in follow-up PR?

File: code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts [hunk 4]
```diff
@@ -142,7 +138,7 @@ beforeEach(() => {
   vi.mocked(WebView.prototype).prepareForStory.mockReturnValue('story-element' as any);
 });
 
-describe('PreviewWeb', () => {
+describe.skip('PreviewWeb', () => {
```
Comment (shilman) [inline code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts:141]: Same q

File: docs/configure/index.md [hunk 5]
```diff
@@ -52,10 +52,9 @@ This configuration file is a [preset](../addons/addon-types.md) and, as such, ha
...
-| `storyStoreV7`        | Configures Storybook to load stories [on demand](#on-demand-story-loading), rather than during boot up (defaults to `true` as of `v7.0`) <br/> `features: { storyStoreV7: true }`             |
-| `buildStoriesJson`    | Generates `index.json` and `stories.json` files to help story loading with the on-demand mode (defaults to `true` when `storyStoreV7` is `true`) <br/> `features: { buildStoriesJson: true }` |
+| Configuration element | Description                                                                                                                                                     |
+| --------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `buildStoriesJson`    | Generates `index.json` and `stories.json` files to help story loading with the on-demand mode (defaults to `true`) <br/> `features: { buildStoriesJson: true }` |
```
Comment (shilman) [inline docs/configure/index.md:57]: Doesn't this PR get rid of `buildStoriesJson`?

**Review by shilman — COMMENTED** [review]
Looks great. Left a few minor comments!

File: code/lib/preview-api/src/modules/preview-web/Preview.tsx (hunk 2 above)
Comment (ndelangen) [reply code/lib/preview-api/src/modules/preview-web/Preview.tsx:166]: Let's just make it `Promise<void>`

File: docs/api/main-config-features.md (hunk 1 above)
Comment (valentinpalkovic) [reply docs/api/main-config-features.md:103]: cc @valentinpalkovic

**Review by valentinpalkovic — APPROVED** [review]

**Pull request merged**